File: InMotionGIT.Common - Copy/DataAcceess/Interface/ITransaction.vb
Namespace DataAccess.Interfaces

    Public Interface ITransaction
        Inherits IDisposable

        Sub Commit()

        Sub Rollback()

        Function SqlCommand(ByVal query As String) As ICommandExecution

        Function StoredProc(ByVal storedProcName As String) As ICommandExecution

    End Interface

End Namespace

File: Common/InMotionGIT.Common/DataAcceess/Interface/ITransaction.cs
using System;

namespace InMotionGIT.Common.DataAccess.Interfaces
{

    public interface ITransaction : IDisposable
    {

        void Commit();

        void Rollback();

        ICommandExecution SqlCommand(string query);

        ICommandExecution StoredProc(string storedProcName);

    }

}